From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Mon, 1 Jul 2019 06:36:30 +0000
Subject: [PATCH] audio_extn: Fix unused parameter warning in utils.c

The unused parameter warning appears if we are not building with
at least one of the following cflags enabled:

* DEV_ARBI_ENABLED
* SOUND_TRIGGER_ENABLED
* AUDIO_LISTEN_ENABLED

  hardware/qcom/audio/hal/audio_extn/utils.c:2522:55: error: unused parameter 'snd_device'
[-Werror,-Wunused-parameter]
  void audio_extn_utils_release_snd_device(snd_device_t snd_device)

Change-Id: I694c683c9bfde60343f0f6ea8d806bc5e24437e6
---
 hal/audio_extn/audio_extn.h | 4 ++++
 hal/audio_extn/utils.c      | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/hal/audio_extn/audio_extn.h b/hal/audio_extn/audio_extn.h
index f35a9d60b0..b8abfd04a4 100644
--- a/hal/audio_extn/audio_extn.h
+++ b/hal/audio_extn/audio_extn.h
@@ -868,7 +868,11 @@ bool audio_extn_utils_is_dolby_format(audio_format_t format);
 int audio_extn_utils_get_bit_width_from_string(const char *);
 int audio_extn_utils_get_sample_rate_from_string(const char *);
 int audio_extn_utils_get_channels_from_string(const char *);
+#if !defined(DEV_ARBI_ENABLED) && !defined(SOUND_TRIGGER_ENABLED) && !defined(AUDIO_LISTEN_ENABLED)
+#define audio_extn_utils_release_snd_device(snd_device) (0)
+#else
 void audio_extn_utils_release_snd_device(snd_device_t snd_device);
+#endif
 bool audio_extn_utils_is_vendor_enhanced_fwk();
 int audio_extn_utils_get_vendor_enhanced_info();
 int audio_extn_utils_get_app_sample_rate_for_device(struct audio_device *adev,
diff --git a/hal/audio_extn/utils.c b/hal/audio_extn/utils.c
index 174e43fa96..9b72e31a2c 100644
--- a/hal/audio_extn/utils.c
+++ b/hal/audio_extn/utils.c
@@ -3312,6 +3312,7 @@ int audio_extn_utils_get_channels_from_string(const char *id_string)
     return -EINVAL;
 }
 
+#if defined(DEV_ARBI_ENABLED) || defined(SOUND_TRIGGER_ENABLED) || defined(AUDIO_LISTEN_ENABLED)
 void audio_extn_utils_release_snd_device(snd_device_t snd_device)
 {
     audio_extn_dev_arbi_release(snd_device);
@@ -3320,6 +3321,7 @@ void audio_extn_utils_release_snd_device(snd_device_t snd_device)
     audio_extn_listen_update_device_status(snd_device,
             LISTEN_EVENT_SND_DEVICE_FREE);
 }
+#endif
 
 int audio_extn_utils_get_license_params(
         const struct audio_device *adev,
